Source/AtomicJS/Javascript/JSScene.cpp

```diff
@@ -24,6 +24,27 @@ static int Node_CreateJSComponent(duk_context* ctx)
     return 1;
 }
 
+static int Node_GetJSComponent(duk_context* ctx)
+{
+	StringHash classNameHash = duk_to_string(ctx, 0);
+	duk_push_this(ctx);
+	Node* node = js_to_class_instance<Node>(ctx, -1, 0);
+
+	PODVector<JSComponent*> jsComponents;
+	node->GetComponents<JSComponent>(jsComponents);
+
+	for (unsigned i = 0; i < jsComponents.Size(); i++)
+	{
+		if (StringHash(jsComponents[i]->GetClassName()) == classNameHash)
+		{
+			js_push_class_object_instance(ctx, jsComponents[i]);
+			break;
+		}
+	}
+
+	return 1;
+}
+
```

Comment:
1) I do realize it would only return the first, I'd think a get JSComponents() that returns all of them would be helpful there (particularly since the named fields will only cover the most recently added component of that type).

2) Yeah you mentioned when we first added this that they could be reached that way, still seemed like something worth keeping though.